lib/docker-toolbelt.coffee

```diff
@@ -102,6 +102,42 @@ Docker::imageRootDir = (image) ->
 					else
 						throw new Error("Unsupported driver: #{dockerInfo.Driver}/")
 	)
+
+EEXIST = (err) -> err.code is 'EEXIST'
+ignore = ->
+
+aufsMountWithDisposer = (target, layerDiffPaths) ->
+	# We try to create the target directory.
+	# If it exists, it's *probably* from a previous run of this same function,
+	# and the mount will fail if the directory is not empty or something's already mounted there.
+	fs.mkdirAsync(target)
+	.catch EEXIST, ignore
+	.then ->
+		appendFromIndex = null
+		options = 'noxino,ro,br='
+		remainingBytes = 4096 - options.length
+		for i in [0...layerDiffPaths.length]
+			nextBranch = layerDiffPaths[i] + '=ro+wh'
+			nextBranch = ':' + nextBranch if i > 0
+			if remainingBytes - nextBranch.length < 0
+				appendFromIndex = i
+				break
+			options += nextBranch
+		execAsync("mount -t aufs -o '#{options}' none #{target}")
+		.then ->
+			if appendFromIndex
+				Promise.mapSeries [appendFromIndex...layerDiffPaths.length], (i) ->
+					execAsync("mount -t aufs -o 'remount,append:#{layerDiffPaths[i]}=ro+wh' none #{target}")
```

Comment:
Can you only append one set at a time? Or does it also have the 4096 limit we can go up to by combining layers?